scheduler/scheduler.py

```diff
@@ -0,0 +1,53 @@
+"""
+scheduler.py
+Module to send the daily reminders using the apscheduler.
+"""
+import logging
+import httpx
+from fastapi import HTTPException
+from apscheduler.schedulers.asyncio import AsyncIOScheduler
+from oauth import create_access_token
+from settings import app_url
+from scheduler.helpers import get_timezones_list
+
+ADMIN_TOKEN = create_access_token(data={"user_id": 1, "user_role": "admin"})
+URL = f"{app_url}:8000/api/v1/users/send_reminders"
+HEADERS = {
+    "Authorization": f"Bearer {ADMIN_TOKEN}"
+}
+scheduler = AsyncIOScheduler()
+
+
+async def send_reminders_to_users(timezone):
+    """
+    Function that performs a POST request to the
+    'send_daily_reminder' endpoint as the admin user.
+    """
+    async with httpx.AsyncClient() as client:
+        try:
+            json_data = {"timezone": timezone}
+            response = await client.post(URL, headers=HEADERS, json=json_data)
+            logging.info("Reminders sent, status code: %s", response.status_code)
+        except HTTPException as error:
+            logging.error("Error while sending reminders: %s", error)
+
+
+async def scheduler_main(job, db_session):
+    """
+    Function to set up the scheduler to execute the
+    send_reminders_to_users function. The refresh_timezones_jobs
+    function is triggered daily to add new timezones as needed
+    after the app's initialization.
+    """
+    async def refresh_timezones_jobs():
+        timezones = await get_timezones_list(db_session)
+        for timezone in timezones:
+            job_id = f"reminder_for_{timezone}"
+            if not scheduler.get_job(job_id):
+                scheduler.add_job(job, 'cron', hour=9, minute=0, timezone=timezone,
+                                  args=[timezone], id=job_id)
```

Comment:
I like that you adding refresh job to keep track if we`ll got users with new timezone.
But maybe we should also check if we need to remove reminder job in cases where all users within tz have disabled notifications